src/index.cpp

```diff
@@ -3255,13 +3123,13 @@ template <typename T, typename TagT, typename LabelT> void Index<T, TagT, LabelT
         std::memcpy(cur_node_offset + sizeof(float), cur_vec, _data_len - sizeof(float));
 
         cur_node_offset += _data_len;
-        uint32_t k = (uint32_t)_final_graph[i].size();
+        uint32_t k = (uint32_t)_graph_store->get_neighbours(i).size();
         std::memcpy(cur_node_offset, &k, sizeof(uint32_t));
-        std::memcpy(cur_node_offset + sizeof(uint32_t), _final_graph[i].data(), k * sizeof(uint32_t));
-        std::vector<uint32_t>().swap(_final_graph[i]);
+        std::memcpy(cur_node_offset + sizeof(uint32_t), _graph_store->get_neighbours(i).data(), k * sizeof(uint32_t));
+        std::vector<uint32_t>().swap(_graph_store->get_neighbours(i));
     }
-    _final_graph.clear();
-    _final_graph.shrink_to_fit();
+    _graph_store->clear_graph();
```

Comment:
instead of two calls to clear_Graph and shrink_to_fit, can we just have one function?

Reply:
I agree. but shrink to fit seems fundamentally different. Right now we are using adjacency list as  our graph data_structure. But lets say in future we have a matrix(or something else) to sore a graph. We need to be able to support shrink to fit and the respective graph store classes will know how to compact the underlying structures.